core/src/main/java/com/segment/analytics/kotlin/core/Events.kt

```diff
@@ -86,17 +86,70 @@ sealed class BaseEvent {
     }
 
     internal suspend fun applyBaseEventData(store: Store) {
-        val system = store.currentState(System::class) ?: return
         val userInfo = store.currentState(UserInfo::class) ?: return
 
         this.anonymousId = userInfo.anonymousId
-        this.integrations = system.integrations ?: emptyJsonObject
+        this.integrations = emptyJsonObject
 
         if (this.userId.isBlank()) {
             // attach system userId if present
             this.userId = userInfo.userId ?: ""
         }
     }
+
+    // Create a shallow copy of this event payload
+    fun <T : BaseEvent> copy(): T {
+        val original = this
+        val copy = when (this) {
+            is AliasEvent -> AliasEvent(userId = this.userId, previousId = this.previousId)
+            is GroupEvent -> GroupEvent(groupId = this.groupId, traits = this.traits)
+            is IdentifyEvent -> IdentifyEvent(userId = this.userId, traits = this.traits)
+            is ScreenEvent -> ScreenEvent(
+                name = this.name,
+                category = this.category,
+                properties = this.properties
+            )
+            is TrackEvent -> TrackEvent(event = this.event, properties = this.properties)
+        }.apply {
+//            type = original.type
+            anonymousId = original.anonymousId
+            messageId = original.messageId
+            timestamp = original.timestamp
+            context = original.context
+            integrations = original.integrations
+            userId = original.userId
+        }
+        @Suppress("UNCHECKED_CAST")
+        return copy as T // This is ok because resultant type will be same as input type
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as BaseEvent
+
+        if (type != other.type) return false
+        if (anonymousId != other.anonymousId) return false
+        if (messageId != other.messageId) return false
+        if (timestamp != other.timestamp) return false
+        if (context != other.context) return false
+        if (integrations != other.integrations) return false
+        if (userId != other.userId) return false
+
+        return true
```

Comment:
not necessary better, but can be simplified as 
```suggestion
        return type == other.type &&
                    anonymousId == other.anonymousId &&
                    messageId == other.messageId &&
                    timestamp == other.timestamp &&
                    context == other.context &&
                    integrations == other.integrations &&
                    userId != other.userId
```

Reply:
i could be wrong, but isn't the current one better for debuggability? if not then I will change it to your suggestion

Reply:
yeah, the current one is actually better for debugability. no need to change it. I believe the compiler will optimize it itself. I'm still live in fear of java word where the compiler warns about this can be simplified 😄.

Reply:
yeah. the current one is better for deubgability. no need to change it. I believe the compiler will optimize it itself. I'm still live in the fear of java word where the compiler warns this can be simplified. 😄